com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -485,6 +523,25 @@ private void SendAllParamsAndState()
             }
         }
 
+        private ulong[] GetTargetClientIds(ulong originClientId)
+        {
+            if (m_ClientIdsExcludingThemselvesCache == null)
+            {
+                m_ClientIdsExcludingThemselvesCache = new Dictionary<ulong, ulong[]>();
+            }
+
+            if (!m_ClientIdsExcludingThemselvesCache.TryGetValue(originClientId, out var ids))
+            {
+                ids = NetworkManager.ConnectedClientsList
```

Comment:
mmm linq is a no no, it allocates way too much

Reply:
mm I see you're caching the result. ok
I'm still a bit wary when we spawn new NetworkAnimators, but that might be a bit too much of a worry. Is there any way to not use linq at all?

Reply:
LINQ is a no-no in hot path, all of these are cached. Would you prefer non-LINQ implementation? Before this change this code was executed every frame, now it only happens if the list of connected clients has changed OR on the first call.

Reply:
what about non-linq cached?

Reply:
cleaned up all LINQ usage, in serialization too